pkg/framework/deployment.go

```diff
@@ -66,6 +82,27 @@ func IsDeploymentAvailable(c client.Client, name, namespace string) bool {
 	return true
 }
 
+// IsDeploymentSynced returns true if provided deployment spec matched one found on cluster
+func IsDeploymentSynced(c client.Client, dep *kappsapi.Deployment, name, namespace string) bool {
+	if err := wait.PollImmediate(RetryShort, WaitLong, func() (bool, error) {
+		d, err := GetDeployment(c, name, namespace)
+		if err != nil {
+			klog.Errorf("Error getting deployment: %v", err)
+			return false, nil
+		}
+		if !reflect.DeepEqual(d.Spec, dep.Spec) {
+			return true, fmt.Errorf("Deployment %q is not updated. Spec is not equal to: %v",
```

Comment:
shouldn't this return false and the error?

Reply:
More like polling here is needless :) Thanks, just returning `false` now. But in case of error it will short circuit polling and return error anyways